### PR TITLE
add human detection model for foreground extraction

### PR DESCRIPTION
**Pull request by studentBC**
This model is suitable for detect people moving on horizontal line

### CODE REVIEW

**Comment by chiahaol** [comment]
Can you pull main to the latest version, solve conflicts and then push again?

**Comment by studentBC** [comment]
> Can you pull main to the latest version, solve conflicts and then push again?

thanks! conflict resolved

File: main.py [hunk 1]
```diff
@@ -155,6 +158,7 @@ def main(args):
     frames = get_frames(cap)
     fg, bg, fgmasks = extract_foreground(frames, args.fgmode)
 
+
     # remove foreground and fill out the removed part in background
```
Comment (chiahaol) [inline main.py]: Remove this extra line

**Comment by chiahaol** [comment]
Lgtm! only some minor issues commented

File: foreground_extraction.py [hunk 2]
```diff
@@ -1,17 +1,21 @@
 import cv2
 import numpy as np
+from imutils.object_detection import non_max_suppression
```
Comment (chiahaol) [inline foreground_extraction.py]: Move this line to next line. We want to separate “imports” and “froms”

File: main.py [hunk 3]
```diff
@@ -175,6 +179,7 @@ def main(args):
     for i in range(2, frame_count):
         rev, nextp = sp.stitch(pp, bg[i])
         panoramas.append(nextp)
+        #cv2.imwrite("./tmp/pframe%d.jpg" % i, nextp)
```
Comment (chiahaol) [inline main.py]: Remove unused code

File: foreground_extraction.py [hunk 4]
```diff
@@ -58,3 +62,36 @@ def get_foreground_mask_gmg(self, frames):
             fgmask = np.where((fgmask == 255), 1, 0).astype('uint8')
             fgmasks.append(fgmask)
         return np.array(fgmasks)
+    #https://pyimagesearch.com/2015/11/09/pedestrian-detection-opencv/
+    #https://thedatafrog.com/en/articles/human-detection-video/
+    def get_foreground_mask_hog(self, frames):
+        # detect people in the image
+        fgmasks = []
+        for frame in tqdm(frames):
+            #rects, weights = self.hog.detectMultiScale(frame, winStride=(8,8) )
```
Comment (chiahaol) [inline foreground_extraction.py:70]: Remove this line

**Review by chiahaol — APPROVED** [review]

File: foreground_extraction.py (hunk 2 above)
Comment (studentBC) [reply foreground_extraction.py]: done

**Pull request merged**